src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
https://wayland.freedesktop.org/libinput/doc/latest/group__event.html#ga8a49a089181960f7536344f7637ea777

^ It is actually possible to differentiate between devices. And it is necessary for device configuration. I guess we need a way to access devices or more callbacks now.